tests/integration/test_integration.py

```diff
@@ -29,209 +31,353 @@
 
-        """
-        assert ops_test.model
-        traefik = ops_test.model.applications[TRAEFIK_APPLICATION_NAME]
-        assert isinstance(traefik, Application)
-        traefik_unit = traefik.units[0]
-        t0 = time.time()
-        while time.time() - t0 < timeout:
-            proxied_endpoint_action = await traefik_unit.run_action(
-                action_name="show-proxied-endpoints"
-            )
-            action_output = await ops_test.model.get_action_output(
-                action_uuid=proxied_endpoint_action.entity_id, wait=30
-            )
-
-            if "proxied-endpoints" in action_output:
-                proxied_endpoints = json.loads(action_output["proxied-endpoints"])
-                return proxied_endpoints[APPLICATION_NAME]["url"]
-            else:
-                logger.info("Traefik did not return proxied endpoints yet")
-            time.sleep(2)
-
-        raise TimeoutError("Traefik did not return proxied endpoints")
-
-    async def run_get_ca_certificate_action(self, ops_test: OpsTest, timeout: int = 60) -> dict:
-        """Run the `get-certificate` on the `vault-k8s` unit.
-
-        Args:
-            ops_test (OpsTest): OpsTest
-            timeout (int, optional): Timeout in seconds. Defaults to 60.
-
-        Returns:
-            dict: Action output
-        """
+    @pytest.mark.abort_on_fail
+    async def test_given_application_is_deployed_when_pod_crashes_then_unit_recovers(
+        self, ops_test: OpsTest, deploy_phase_1
+    ):
         assert ops_test.model
-        self_signed_certificates_unit = ops_test.model.units[
-            f"{SELF_SIGNED_CERTIFICATES_APPLICATION_NAME}/0"
-        ]
-        assert isinstance(self_signed_certificates_unit, Unit)
-        action = await self_signed_certificates_unit.run_action(
-            action_name="get-ca-certificate",
+        unit = ops_test.model.units[f"{APPLICATION_NAME}/1"]
+        assert isinstance(unit, Unit)
+        k8s_namespace = ops_test.model.name
+        crash_pod(name=f"{APPLICATION_NAME}-1", namespace=k8s_namespace)
+        await ops_test.model.wait_for_idle(
+            apps=[APPLICATION_NAME],
+            status="active",
+            timeout=1000,
+            wait_for_exact_units=NUM_VAULT_UNITS,
         )
-        return await ops_test.model.get_action_output(action_uuid=action.entity_id, wait=timeout)
 
     @pytest.mark.abort_on_fail
-    @pytest.fixture(scope="module")
-    async def deploy_prometheus(self, ops_test: OpsTest) -> None:
-        """Deploy Prometheus.
-
-        Args:
-            ops_test: Ops test Framework.
-        """
+    async def test_given_application_is_deployed_when_scale_up_then_status_is_active(
+        self, ops_test: OpsTest, deploy_phase_1
+    ):
         assert ops_test.model
-        await ops_test.model.deploy(
-            "prometheus-k8s",
-            application_name=PROMETHEUS_APPLICATION_NAME,
-            trust=True,
+        num_units = 5
```

Comment:
Would it make sense for this to be `NUM_VAULT_UNITS + 2`? Incase `NUM_VAULT_UNITS` is ever changed to 5, for example?

Reply:
I forgot I changed this for testing the memory usage, thanks for catching it.